Sources/PackageLoading/Diagnostics.swift

```diff
@@ -87,6 +87,39 @@ extension Diagnostic.Message {
             resource file in the resources bundle
             """)
     }
+
+    static func copyConflictWithLocalizationDirectory(path: RelativePath, targetName: String) -> Self {
+        .error("resource '\(path)' in target '\(targetName)' conflicts with other localization directories")
+    }
+
+    static func missingDevelopmentRegion() -> Self {
+        .error("missing manifest property 'developmentRegion'; it is required in the presence of localized resources")
+    }
+
+    static func localizationAmbiguity(path: RelativePath, targetName: String) -> Self {
+        .error("""
+            resource '\(path)' in target '\(targetName)' is in a localization directory and has an explicit \
+            localization declaration; choose one or the other to avoid any ambiguity
+            """)
+    }
+
+    static func localizedAndUnlocalizedVariants(resource: String, targetName: String) -> Self {
+        .warning("""
+            resource '\(resource)' in target '\(targetName)' has both localized and un-localized variants; the \
+            localized variants will never be chosen
+            """)
+    }
+
+    static func missingDevelopmentRegionResource(
+        resource: String,
+        targetName: String,
+        developmentRegion: String
+    ) -> Self {
+        .warning("""
```

Comment:
Any reason to not make this an error as well?

Reply:
Because it won't break anything: Foundation will just default to "en" anyway. The idea was that this was a nice-to-have warning to users.

Reply:
@neonichu What do you think?

Reply:
The [bundle search pattern documentation](https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFBundles/AccessingaBundlesContents/AccessingaBundlesContents.html#//apple_ref/doc/uid/10000123i-CH104-SW7) does not mention the fallback to `en`. Since the package may not even support `en` at all, it could lead to crashes for API’s with non‐optional return values such as `NSLocalizedString`. (Right?) I would say make this an error at least if the package doesn’t support `en`. For simplicity and for consistency with the documentation, it would be simplest to just make it an error regardless.

Reply:
Or maybe just a warning for the sake of the API’s that _do_ have optional return values (and with the `localization:` parameter)? I don’t know.